File: BookStoreNetReact.Application/Helpers/Pagination.cs
namespace BookStoreNetReact.Application.Helpers
{
    public class Pagination
    {
        public required int PageSize { get; set; } = 6;
        public required int PageIndex { get; set; } = 1;
        public required int TotalCount { get; set; }
        public required int TotalPages { get; set; }
    }
}

File: client/src/app/models/pagination.ts
export interface FilterParams {
    pageIndex: number;
    pageSize: number;
    sort: string;
}
export interface MetaData {
    pageIndex: number;
    totalPages: number;
    pageSize: number;
    totalCount: number;
}

export class PaginatedResponse<T> {
    items: T;
    metaData: MetaData;

    constructor(items: T, metaData: MetaData) {
        this.items = items;
        this.metaData = metaData;
    }
}
